bigframes/core/compile/compiled.py

```diff
@@ -355,6 +346,9 @@ def _to_ibis_expr(
         return table
 
     def filter(self, predicate: ex.Expression) -> UnorderedIR:
+        if any(map(is_window, map(self._get_ibis_column, predicate.unbound_variables))):
```

Comment:
Is this a fix for something? If so, maybe worth a mention in the changelog and corresponding test?

Reply:
This is not a fix, previously we would proactively reproject after windowing. With this new implementation, instead, we are maximally deferring, so need to check later on wherever windowed expressions are not allowed.